File: dynamicpdf-csharp-examples/Examples/data/ProductCategoryData.cs
using DynamicPDFCoreSuite.Examples;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.IO;

    class ProductCategoryData
    {
    public class Product
    {
        public int ProductID { get; set; }
        public string ProductName { get; set; }
        public string QuantityPerUnit { get; set; }
        public bool Discontinued { get; set; }
        public decimal UnitPrice { get; set; }
    }

    public class ProductCategory
    {
        public string Name { get; set; }
        public List<Product> Products { get; set; }
    }

    public class Root
    {
        public List<ProductCategory> ProductsByCategory { get; set; }
    }

    public static List<ProductCategory> GetProductCategoryObjects()
    {
        string jsonFilePath = Util.GetPath("Resources/Data/subreport.json");
        string jsonData = File.ReadAllText(jsonFilePath);
        var productData = JsonConvert.DeserializeObject<ProductCategoryData.Root>(jsonData);
        //DisplayParseData(productData);
        return productData.ProductsByCategory;

    }

    private static void DisplayParseData(Root productData)
    {
        foreach (var category in productData.ProductsByCategory)
        {
            Console.WriteLine($"Category: {category.Name}");
            foreach (var product in category.Products)
            {
                Console.WriteLine($" - {product.ProductName}, Price: ${product.UnitPrice}");
            }
        }

    }
}

File: dynamicpdf-vbnet-examples/Examples/Data/ProductCategoryData.vb
Imports System
Imports System.Collections.Generic
Imports System.IO
Imports Newtonsoft.Json

Public Class ProductCategoryData

    Public Class Product
        Public Property ProductID As Integer
        Public Property ProductName As String
        Public Property QuantityPerUnit As String
        Public Property Discontinued As Boolean
        Public Property UnitPrice As Decimal
    End Class

    Public Class ProductCategory
        Public Property Name As String
        Public Property Products As List(Of Product)
    End Class

    Public Class Root
        Public Property ProductsByCategory As List(Of ProductCategory)
    End Class

    Public Shared Function GetProductCategoryObjects() As List(Of ProductCategory)
        Dim jsonFilePath As String = Util.GetPath("Resources/Data/subreport.json")
        Dim jsonData As String = File.ReadAllText(jsonFilePath)
        Dim productData As Root = JsonConvert.DeserializeObject(Of Root)(jsonData)
        ' DisplayParseData(productData)
        Return productData.ProductsByCategory
    End Function

    Private Shared Sub DisplayParseData(ByVal productData As Root)
        For Each category In productData.ProductsByCategory
            Console.WriteLine($"Category: {category.Name}")
            For Each product In category.Products
                Console.WriteLine($" - {product.ProductName}, Price: ${product.UnitPrice}")
            Next
        Next
    End Sub

End Class
